src/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -0,0 +1,675 @@
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="uint"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        [CLSCompliant(false)]
+        public JsonArray(IEnumerable<uint> values) : this()
+        {
+            foreach (uint value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="ulong"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        [CLSCompliant(false)]
+        public JsonArray(IEnumerable<ulong> values) : this()
+        {
+            foreach (ulong value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="decimal"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        public JsonArray(IEnumerable<decimal> values) : this()
+        {
+            foreach (decimal value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets the element at the specified index.
+        /// </summary>
+        /// <param name="idx">The zero-based index of the element to get or set.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   Index is less than 0.
+        /// </exception>
+        public JsonNode this[int idx]
+        {
+            get
+            {
+                if (idx < 0 || idx >= _list.Count)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                return _list[idx];
+            }
+            set
+            {
+                if (idx < 0 || idx >= _list.Count)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                _list[idx] = value;
+            }
+        }
+
+        /// <summary>
+        ///   Adds the specified <see cref="JsonNode"/> value to the JSON array.
+        /// </summary>
+        /// <param name="value">The value to add.</param>
+        /// <remarks>Null value is allowed and represents a null JSON node.</remarks>
+        public void Add(JsonNode value) => _list.Add(value);
+
+        /// <summary>
+        ///   Adds the specified value as a <see cref="JsonString"/> to the JSON array.
+        /// </summary>
+        /// <param name="value">The value to add.</param>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided value is null.
```

Comment:
Nulls should work here, producing the JSON null value.

Reply:
Ok, I will fix it.